docs/environment.md

```diff
@@ -0,0 +1,186 @@
+<!---
+Licensed under Creative Commons Attribution 4.0 International License
+https://creativecommons.org/licenses/by/4.0/
+--->
+# Private Data Objects Environment Variables
+
+Instructions in this document assume the environment variable
+`PDO_SOURCE_ROOT` points to the PDO source directory.
+
+PDO uses a number of environment variables to control build,
+installation and operation. While PDO should build and run with only the
+default values, three variables are commonly set to reflect specifics of
+the installation:
+
+  * [`PDO_INSTALL_ROOT`](#pdo_install_root) -- the path to the directory where PDO is installed
+  * [`PDO_LEDGER_URL`](#pdo_ledger_url) -- the URL for the Sawtooth ledger
+  * [`PDO_STL_KEY_ROOT`](#pdo_stl_key_root) -- the path to a directory containing Sawtooth keys
+
+In addition, if you run in SGX HW mode you will generally define
+`PDO_SGX_KEY_ROOT` as well. See below for information on these variables
+and others you could override from defaults.
+
+<!-- -------------------------------------------------- -->
+<!-- -------------------------------------------------- -->
+## Common Configuration Script
+
+The script
+[build/common-config.sh](../build/common-config.sh)
+can be used to set values for all of the environment variables that are
+used in the build, installation & execution process.
+
+The default usage of this script is to be sourced. For example, local
+configuration file may be constructed as:
+
+```bash
+   export PDO_INSTALL_ROOT=${PDO_SOURCE_ROOT}/build/_dev
+   export PDO_STL_KEY_ROOT=${PDO_INSTALL_ROOT}/opt/pdo/etc/keys/sawtooth
+   export PDO_LEDGER_URL=http://127.0.0.1:8008
+```
```

Comment:
The default PDO_LEDGER_URL=http://127.0.0.1:8008 is confusing since this suggests that sawtooth is installed as a separate process in the same VM. This is after we say that sawtooth needs ubuntu 16, PDO needs 18, and that we recommend them to be on separate VMs. 

Summary: Does the user need to set up two VMs  to see a demo of PDO?  It will be great if not.

Reply:
its an example.

Reply:
Ok, but this is a misleading/invalid example. you never set it to 127.0.0.1:8008 on the machine where you install PDO since you always want it on a different machine

Reply:
i don't believe there is anything better that we could put there. That is, this is the best example we can construct.